pkg/jobs/scheduled_job.go

```diff
@@ -462,7 +464,18 @@ func (s scheduledJobStorageTxn) Create(ctx context.Context, j *ScheduledJob) err
 	if j.rec.ScheduleID != 0 {
 		return errors.New("cannot specify schedule id when creating new cron job")
 	}
-
+	if j.rec.ScheduleDetails.ClusterID.Equal(uuid.UUID{}) {
+		return errors.New("scheduled job created without a cluster ID")
+	}
+	if j.rec.ScheduleDetails.CreationClusterVersion.Equal(clusterversion.ClusterVersion{}) {
+		return errors.New("scheduled job created without a cluster version")
+	}
+	if j.rec.ScheduleDetails.CreationClusterVersion.Less(clusterversion.ByKey(clusterversion.V23_2)) {
```

Comment:
is this strictly necessary? Won't older binaries just not have those fields on the proto they unmarshal the column into. What would make me feel confident here is a mixed version logic test that:

- creates a schedule on master and loads it on 23.1
- creates a schedule on 23.1 and loads it on master

Reply:
good callout. I added a mixed version test for your perusal. 

I version gated the _writing_ of the clusterID and version because in a mixed version state, older nodes _could_ overwrite these new fields. e.g. `ALTER BACKUP SCHEDULE ...  SET SCHEDULE OPTION on_previous_running = 'wait';`  from an old node in a mixed version state would overwrite these new fields.  It just seems cleaner to me for these fields to only populate once we're fully upgraded.